src/ResourceManager/HDInsight/Commands.HDInsight/ManagementCommands/AddAzureHDInsightSecurityProfile.cs

```diff
@@ -0,0 +1,98 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Commands.HDInsight.Commands;
+using Microsoft.Azure.Commands.HDInsight.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet(
+        VerbsCommon.Add,
+        Constants.CommandNames.AzureHDInsightSecurityProfile),
+    OutputType(
+        typeof(AzureHDInsightSecurityProfile))]
+    public class AddAzureHDInsightSecurityProfile : HDInsightCmdletBase
+    {
+        private readonly AzureHDInsightSecurityProfile _securityProfile;
+
+        #region Input Parameter Definitions
+
+        [Parameter(Position = 0,
+        Mandatory = true,
+        ValueFromPipeline = true,
+        HelpMessage = "The HDInsight cluster configuration to use when creating the new cluster.")]
+        public AzureHDInsightConfig Config { get; set; }
+
+        [Parameter(Position = 1,
+            Mandatory = true,
+            HelpMessage = "The Domain of the secure cluster.")]
+        public string Domain
+        {
+            get { return _securityProfile.Domain; }
+            set { _securityProfile.Domain = value; } 
+        }
+
+        [Parameter(Position = 2,
+            Mandatory = true,
+            HelpMessage = "Credential of the domain user. Username should be in user principal format. Eg. sampleuser@contoso.com")]
```

Comment:
"Credential of the domain user. Username should be in user principal name format, e.g. sampleuser@contoso.com"